### DISCUSSION TITLE
joinedload not working with FK fields generated by Django migration

### DISCUSSION THREAD

**Discussion by overide** [sqlalchemy/sqlalchemy#6627] [Usage Questions] [answered]
I'm using `automap` with tables that are originally generated by Django migration. Everything works fine but I cannot use `joinedload` and other relationship loading options to pre-fetch the related fields. I get the following error - 
```
sqlalchemy.orm.exc.LoaderStrategyException: Can't apply "joined loader" strategy to property "inspection_image.project_id", which is a "column property"; this loader strategy is intended to be used with a "relationship property".
```
Model looks like this in Django - 
```python
class Image(models.Model):
    project = models.ForeignKey(Project, on_delete=models.CASCADE)
```

Query looks like this - 
```python
Base = automap_base()
Base.prepare(engine, reflect=True)
Image = Base.classes.inspection_image
query = db.select(Image).options(joinedload(Image.project_id))
```
When I inspected the auto mapped table columns, `project_id` field type is ` BIGINT` but it's defined as  `ForeignKey` in Django models so I was expecting that automap should automatically detect the relationship, as it's mention in documentation - https://docs.sqlalchemy.org/en/14/orm/extensions/automap.html#module-sqlalchemy.ext.automap

PS: 
1. There are many Django models so I don't want to explicitly define classes for each one of them
2. Very new to SqlAlchemy, so please let me know if there is any better way to do the same

**Comment by zzzeek** [comment]
automap uses [name_for_scalar_relationship](https://docs.sqlalchemy.org/en/14/orm/extensions/automap.html#sqlalchemy.ext.automap.name_for_scalar_relationship) to determine the name of this relationship, which in this case would be "project", so:


    db.select(Image).options(joinedload(Image.project))

**Reply by overide** [reply]
I've tried `project` as well and get an error that there is no such field
```
AttributeError: type object 'inspection_image' has no attribute 'project'
```

Note: Django renames the ForeignKey fields and appends `_id` to the field name while creating that column in DB table

**Comment by overide** [comment, marked as answer]
Okay, so the relationship name was different for FKs, the generated relationship name was like - `<django-app-label>_<related-model-name>`. In my case the - 

App label for `Project` model is - `project`
Model name for Project model is - `project`

So final relationship name is - `project_project`

You can use the following code to get all the relationship name for your class if you are not sure about name - 
```python
from sqlalchemy.inspection import inspect
from sqlalchemy.ext.automap import automap_base

Base = automap_base()
Base.prepare(engine, reflect=True)
MyClass = Base.classes.your_table_name # usually `<app-label>_<model_name_lowercase>`

for x in inspect(MyClass).relationships.items():
    print(x[0])
```
